src/main/java/com/epam/ta/reportportal/core/events/handler/DefectTypeDeletedHandler.java

```diff
@@ -0,0 +1,90 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.ta.reportportal.core.events.handler;
+
+import com.epam.ta.reportportal.core.analyzer.LogIndexer;
+import com.epam.ta.reportportal.core.analyzer.client.AnalyzerServiceClient;
+import com.epam.ta.reportportal.core.analyzer.impl.AnalyzerStatusCache;
+import com.epam.ta.reportportal.core.events.activity.DefectTypeDeletedEvent;
+import com.epam.ta.reportportal.dao.LaunchRepository;
+import com.epam.ta.reportportal.dao.ProjectRepository;
+import com.epam.ta.reportportal.entity.project.Project;
+import com.epam.ta.reportportal.exception.ReportPortalException;
+import com.epam.ta.reportportal.ws.model.ErrorType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.transaction.event.TransactionalEventListener;
+
+import java.util.List;
+
+import static com.epam.ta.reportportal.commons.Predicates.equalTo;
+import static com.epam.ta.reportportal.commons.validation.BusinessRule.expect;
+import static com.epam.ta.reportportal.core.analyzer.impl.AnalyzerUtils.getAnalyzerConfig;
+
+/**
+ * @author <a href="mailto:ihar_kahadouski@epam.com">Ihar Kahadouski</a>
+ */
+@Component
+@Transactional(propagation = Propagation.REQUIRES_NEW)
+public class DefectTypeDeletedHandler {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(DefectTypeDeletedHandler.class);
+
+	private final AnalyzerStatusCache analyzerStatusCache;
+
+	private final AnalyzerServiceClient analyzerServiceClient;
+
+	private final LaunchRepository launchRepository;
+
+	private final LogIndexer logIndexer;
+
+	private final ProjectRepository projectRepository;
+
+	@Autowired
+	public DefectTypeDeletedHandler(AnalyzerStatusCache analyzerStatusCache, AnalyzerServiceClient analyzerServiceClient,
+			LaunchRepository launchRepository, LogIndexer logIndexer, ProjectRepository projectRepository) {
+		this.analyzerStatusCache = analyzerStatusCache;
+		this.analyzerServiceClient = analyzerServiceClient;
+		this.launchRepository = launchRepository;
+		this.logIndexer = logIndexer;
+		this.projectRepository = projectRepository;
+	}
+
+	@TransactionalEventListener
+	public void handleDfectTypeDeleted(DefectTypeDeletedEvent event) {
+		Project project = projectRepository.findById(event.getProjectId())
+				.orElseThrow(() -> new ReportPortalException(ErrorType.PROJECT_NOT_FOUND, event.getProjectId()));
+
+		if (analyzerServiceClient.hasClients()) {
+			expect(
```

Comment:
okey
if there will be running analyzer and the delete defect type event comes
you forbid updating indexing 
results:
analyzer finishes running
defect type is deleted and index is not updated

probably you need to implement some kind of retry logic to wait analyzing finish.
spring retry for example